src/pages/index.js

```diff
@@ -0,0 +1,282 @@
+      // Update the UI accordingly
+      likeButton.classList.toggle("card__like-btn_active", !isLiked);
+      const likeCounter = likeButton
+        .closest(".card")
+        .querySelector(".card__like-count");
+      if (likeCounter) {
+        likeCounter.textContent = updatedCard.likes.length;
+      }
+    })
+    .catch((err) => {
+      console.error("Error updating like status:", err);
+    });
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+}
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+}
+function handleEditFormSubmit(evt) {
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  setButtontext(submitBtn, false, "Saving...");
+  evt.preventDefault();
+  api
+    .editUserInfo({ name: editModalNameInput.value, about: editDescriptionInput.value, })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // submitBtn.textContent = "Save";
+      setButtontext(submitBtn, true);
+    });
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+let selectedCard, selectedCardId;
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  api.addCard(inputValues).then((res) => {
+    console.log(res);
+    const cardEl = getCardElement(res);
+    cardsList.prepend(cardEl);
+    cardForm.reset();
+    disableButton(cardSubmitBtn,settings);
+    closeModal(cardModal);
+  });
+}
+
+function handleDeleteCardSubmit(evt) {
+  evt.preventDefault();
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error);
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  api
+    .editAvatarInfo({ avatar: avatarInput.value })
+    .then((data) => {
+      avatarImage.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editDescriptionInput.value = profileDescription.textContent;
+  resetValidation(editFormElement, settings);
+  openModal(editModal);
+});
+editProfileClosedBtn.addEventListener("click", () => {
+  closeModal(editModal);
+});
+cardModalBtn.addEventListener("click", () => {
+  openModal(cardModal);
+});
+editFormElement.addEventListener("submit", handleEditFormSubmit);
+cardForm.addEventListener("submit", handleAddCardSubmit);
+
+avatarSubmitBtn.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
```

Comment:
* The `cancel` button doesn't close the `delete` modal https://snipboard.io/jS1hpR.jpg